File: ejercicio2.cpp
#include <iostream>
#include <map>
#include <string>
using namespace std;

class Producto {
public:
    string nombre;
    float precio;

    Producto() {
        nombre = "";
        precio = 0;
    }

    Producto(string n, float p) {
        nombre = n;
        precio = p;
    }
};

int main() {
    map<string, Producto> productos;
    
    productos["A001"] = Producto("Lapiz", 8.50);
    productos["A002"] = Producto("Cuaderno", 35.00);
    productos["A003"] = Producto("Pluma", 12.00);

    string codigo;
    cout << "Codigo a buscar: ";
    cin >> codigo;

    if (productos.find(codigo)!= productos.end()) {
        Producto producto = productos[codigo];
        cout << "Producto: " << producto.nombre << endl;
        cout << "Precio: $" << producto.precio << endl;
    } else {
        cout << "Codigo no encontrado" << endl;
    }

    return 0;
}

File: ejercicio2.py
class Producto:
    def __init__(self, nombre, precio):
        self.nombre = nombre
        self.precio = precio

productos = {
    "A001": Producto("Lapiz", 8.50),
    "A002": Producto("Cuaderno", 35.00),
    "A003": Producto("Pluma", 12.00)
}

codigo = input("Codigo a buscar: ")

if codigo in productos:
    producto = productos[codigo]
    print("Producto:", producto.nombre)
    print("Precio: $", producto.precio)
else:
    print("Codigo no encontrado")
